### DISCUSSION TITLE
h

### DISCUSSION THREAD

**Discussion by ChandlerAx** [Taureon/aps-plus-plus#235] [General]
is this legal to host

**Comment by Taureon** [comment]
this repository in of itself does not violate copyright laws, neither would hosting
what are you worried about?

**Comment by ChandlerAx** [comment]
> this repository in of itself does not violate copyright laws, neither would hosting what are you worried about?

If you run it in production will it violate any terms of arras.io, diep.io including copyrights?

**Comment by Taureon** [comment]
there is no reason it should. you can't exactly copyright the idea of "geometrically simplistic players shooting each other"
plus, arras.io never got in trouble for ripping off diep.io, so i don't see any legal issues coming for an open source project that aims to be arras.io but better

**Comment by ChandlerAx** [comment]
> there is no reason it should. you can't exactly copyright the idea of "geometrically simplistic players shooting each other" plus, arras.io never got in trouble for ripping off diep.io, so i don't see any legal issues coming for an open source project that aims to be arras.io but better

Cool, do you know how much players 1 cpu thread can handle? and also i have tried aps++ and it doesnt run smoothly like the real arras in rendering performance any solutions i have a RTX 3060 and i see mini lagg spikes + bullets doesnt go smooth

**Comment by Taureon** [comment]
aps++ is bad in rendering stuff simply because we use canvas2d, and also cuz text rendering is taxing

**Comment by ChandlerAx** [comment]
> aps++ is bad in rendering stuff simply because we use canvas2d, and also cuz text rendering is taxing

I hope you can view the performance and update it if necessary.
Also is there a way to separate the client and server as I am going to create a server and game mode selector, my players latency is huge so we have 3 locations and thanks for making aps++, we have 2000+ daily players

**Comment by TheGreenSquare** [comment]
> Also is there a way to separate the client and server as I am going to create a server and game mode selector

we'll get around to it

**Comment by JBX2060** [comment]
What game are you making? BTW just make sure you open source your own version because the license is GPL.

**Comment by NightFuryToothless** [comment]
> Also is there a way to separate the client and server as I am going to create a server and game mode selector, my players latency is huge so we have 3 locations and thanks for making aps++, we have 2000+ daily players

Yes, there's. [A simple multiserver thingy](https://github.com/Taureon/aps-plus-plus/commit/cdb4a862b91f8dbb680f62003d6acd84c70f3977) - the commit that adds a basic multi server support.

*I will update it on the same [branch](https://github.com/Taureon/aps-plus-plus/tree/toothless-to-main), if I ever will*

**Comment by ChandlerAx** [comment]
Can you tell more about how setup multi server support + how make the server uses all cpu cores instead of 1

**Reply by Taureon** [reply]
> Can you tell more about how setup multi server support

there is a browser branch which gives a better ui for browsing servers

> how make the server uses all cpu cores instead of 1

you basically launch multiple instances of aps++ with different configuration files (INCLUDING different port for each instance)

**Comment by NightFuryToothless** [comment]
> Can you tell more about how setup multi server support

Copy the changes I made in this commit. It divides a client and a server into 2 different instances that can run separately, it also let's you use more then one game server.
There's also taureon version of server browser in [browser](https://github.com/Taureon/aps-plus-plus/tree/browser) branch, but it's a little bit unfinished, I might release after version of it with a browser that looks like an arras.io one.
